src/Microsoft.SqlTools.ServiceLayer/Utility/SqlScriptFormatter.cs

```diff
@@ -176,6 +176,83 @@ public static string UnwrapLiteral(string literal)
             return literal;
         }
 
+        public static string[] DecodeMultipartIdenfitier(string multipartIdentifier)
+        {
+            StringBuilder sb = new StringBuilder();
+            List<string> namedParts = new List<string>();
+            bool insideBrackets = false;
+            bool bracketsClosed = false;
+            for (int i = 0; i < multipartIdentifier.Length; i++)
+            {
+                char iChar = multipartIdentifier[i];
+                if (insideBrackets)
+                {
+                    if (iChar == ']')
```

Comment:
Do we need to escape `[` instead of `]`? Looking here under "Using Wildcard Characters As Literals":
https://docs.microsoft.com/en-us/sql/t-sql/language-elements/like-transact-sql